library/src/schemas/variant/variantAsync.ts

```
@@ -21,6 +21,66 @@ import type {
+      discriminatorSchema.type === 'enum' ||
+      discriminatorSchema.type === 'picklist'
+    ) {
+      // @ts-expect-error
+      values = discriminatorSchema.options;
+    } else {
+      // Non-enumerable discriminator (e.g. optional, union, custom)
+      return null;
+    }
+    for (const value of values) {
+      // `NaN` would match under SameValueZero but never under `===`
+      if (value !== value) {
```

Biome (2.5.1)

[error] This comparison uses the same expression on both sides.

(lint/suspicious/noSelfCompare)